lib/bgs/vnp_relationships.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module BGS
+  class VnpRelationships
+    def initialize(proc_id:, veteran:, dependents:, user:)
+      @user = user
+      @veteran = veteran
+      @proc_id = proc_id
+      @dependents = dependents
+    end
+
+    def create
+      spouse_marriages, vet_dependents = @dependents.partition do |dependent|
+        dependent[:type] == 'spouse_marriage_history'
+      end
+
+      spouse = @dependents.find { |dependent| dependent[:type] == 'spouse' }
+
+      spouse_marriages.each do |dependent|
+        bgs_service.create_relationship(
+          vnp_relationship.params_for_686c(spouse[:vnp_participant_id], dependent)
+        )
+      end
+
+      vet_dependents.each do |dependent|
+        bgs_service.create_relationship(
+          vnp_relationship.params_for_686c(@veteran[:vnp_participant_id], dependent)
+        )
+      end
+    end
```

Comment:
Is it intentional that `spouse_marriages` get excluded from what this method is returning?

[In one of your specs further down](https://github.com/department-of-veterans-affairs/vets-api/pull/4618/files#diff-9b2c0648a6001a96fc7dab4f182e978eR43-R45), it looks like you're expecting that all created dependents are being returned by this `create` method.

Reply:
Good question. I have refactored the tests to convey intent.